packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -458,15 +466,28 @@ function useRestoreFocus(scopeRef: RefObject<HTMLElement[]>, restoreFocus: boole
         document.removeEventListener('keydown', onKeyDown, true);
       }
 
-      if (restoreFocus && nodeToRestore && isElementInScope(document.activeElement, scopeRef.current)) {
-        requestAnimationFrame(() => {
-          if (document.body.contains(nodeToRestore)) {
-            focusElement(nodeToRestore);
-          }
-        });
+      if (nodeToRestore) {
+        if (restoreFocus && isElementInScope(document.activeElement, scopeRef.current)) {
+          requestAnimationFrame(() => {
+            // If the nodeToRestore is not the last in the nodeToRestoreArray,
+            // do nothing because a later focusScope is open and should take precedence.
+            if (nodeToRestoreArray.indexOf(nodeToRestore) !== nodeToRestoreArray.length - 1) {
+              return;
+            }
```

Comment:
Looks like this line is responsible for the behavior I noted here: https://github.com/adobe/react-spectrum/pull/2445#pullrequestreview-977644837. The above makes an assumption that the later FocusScope isn't affected by the unmounting of a preceding FocusScope right? Can we make that assumption?

Is this block of code an optimization or was there a case covered here?